native/src/components/CustomHeaderButtons.tsx

```diff
@@ -1,36 +1,27 @@
 import React, { ReactElement, ReactNode } from 'react'
 import { useTranslation } from 'react-i18next'
 import { StyleProp } from 'react-native'
-import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
 import {
   defaultOnOverflowMenuPress,
   HeaderButton,
   HeaderButtons,
   OnOverflowMenuPressParams,
   OverflowMenu,
 } from 'react-navigation-header-buttons'
-import { useTheme } from 'styled-components'
 
-import { LanguageIcon, SearchIcon } from '../assets'
-import { isRTL } from '../constants/contentDirection'
+import { LanguageIcon, MenuIcon, SearchIcon } from '../assets'
+import Icon from './base/Icon'
 
-const Icon = (props: { name: string; style: StyleProp<SVGImageElement> }): ReactElement => {
-  const { name } = props
-  if (name === 'search') {
-    return <SearchIcon {...props} />
-  }
-  if (name === 'language') {
-    return <LanguageIcon scaleX={isRTL() ? -1 : 1} {...props} />
-  }
-  return <MaterialIcon {...props} name={name} />
-}
+const HeaderIcon = ({ name, ...props }: { name: string; style: StyleProp<SVGImageElement> }): ReactElement => (
```

Comment:
Does this really what it should do? Now LanguageIcon is somehow the fallback. I think it would be more stable code, if we have a proper fallback.

Reply:
Hmmm. In general true, just not sure what to use as a fallback and whether it actually makes sense to just introduce a new fallback icon that is then never used. I think it should be quite easy to spot if a new header icon would be added that the wrong icon is rendered. Open for suggestions here though.

Reply:
We could also add a throw Error block if the name isn't either language or search

Reply:
Yes I think this would be a better solution :+1: